python/context/ball_trajectories.py

```diff
@@ -274,6 +306,233 @@ def overwrite(
+                steps = _trim(steps)
+
+                # sometimes there is the same ball id for two steps in a row,
+                # i.e. new ball information did not come up yet during recording
+                # setting these ball id to -1
+                _ball_id_duplicates(steps)
+                
+                # if the ball is not detected at the end,
+                # also removing the corresponding steps
+                steps.reverse()
+                steps = _trim(steps)
+                steps.reverse()
+
+                init_steps = copy.deepcopy(steps)
+                
+                # computing missing position (via
+                # linear interpolation)
+                last_detection = 0
+                detecting = True
+                for index, step in enumerate(steps):
+                    ball = step[1]
+                    if ball.ball_id >= 0:
+                        if not detecting:
+                            values = _interpolation(
+                                index - last_detection,
+                                steps[last_detection][1].position,
+                                ball.position,
+                            )
+                            for index_, position in enumerate(values):
+                                steps[index_ + last_detection][1].position = position
+                        last_detection = index
+                        detecting = True
+                    else:
+                        detecting = False
+
+                # after computing the position, fixing the velocity
+                _recompute_ball_velocity(steps)
+
+                return steps
+
+            # reading the file
+            with open(o80_robot_ball_file, "r") as f:
+                lines = f.readlines()
+
+            # parsing the content of the file
+            steps: typing.List[Step]
+            steps = [_readline(line) for line in lines]
+
+            # fixing the steps for which the ball
+            # was not detected (ball_id < 0)
+            # (fixing: performing linear interpolation to fill the gaps)
+            steps = _fix_undetected_ball(steps)
+
+            # casting to stamped trajectory
+            start_time = steps[0][0]
+            time_stamps = np.array([(step[0] - start_time) * 1e-3 for step in steps])
+            positions = np.array(
+                [list(step[1].position) + list(step[2].position) for step in steps]
+            )
+
+            to_duration_trajectory((time_stamps, positions))
+                
+            return time_stamps, positions
+
+        def _read_folder(tennicam_path: pathlib.Path) -> StampedTrajectories:
+            """
+            List all the file in tennicam_path that have the tennicam_ prefix,
+            parse them and returns the corresponding list of stamped trajectories.
+            """
+            files = _list_files(o80_robot_ball_path, prefix="o80_robot_ball_")
```

Comment:
There seems to be a mismatch between the prefix mentioned in the docstring and the one actually used.